### PR TITLE
Fix/recommendation text visible

### LINKED ISSUE #311: [BUG] Recommendation text visible when there are no recommendation
**Describe the bug**
See recommendation text when there are no recommendations. "คุณอาจสนใจวิชาเหล่านี้"

**To Reproduce**
1. Activate google optimize [random varaint](https://www.googleoptimize.com/opt_set_cookie?uiv2&id=OPT-P9XQVG8&gtm_auth=eGB0WZl6th2-1s0WzM0uQg&gtm_debug&gtm_experiment=OPT-P9XQVG8_OPT-54XQL%241&gtm_preview=opt_preview-slim&redirect=https%3A%2F%2Fwww.google-analytics.com%2Fgtm%2Fset_cookie%3Fuiv2%26id%3DOPT-P9XQVG8%26gtm_auth%3DeGB0WZl6th2-1s0WzM0uQg%26gtm_debug%26gtm_experiment%3DOPT-P9XQVG8_OPT-54XQL%25241%26gtm_preview%3Dopt_preview-slim%26redirect%3Dhttps%253A%252F%252Foptimize.google.com%252Foptimize%252Fsharepreview%253Fid%253DOPT-P9XQVG8%2526gtm_experiment%253DOPT-P9XQVG8_OPT-54XQL%2525241%2526url%253Dhttps%25253A%25252F%25252Fcugetreg.com%25252FS%25252Fcourses%2526opt_experiment_name%253DRecommendation%252520Text%2526opt_variation_name%253DRANDOM%2526slim%253Dtrue%2526container_name%253DCU%252520Get%252520Reg%26optimize_editor&optimize_editor) (Also has a problem on Cosine variant) and go to course page 
2. Add some courses
3. Refresh page

**Expected behavior**
No course recommendation

**Screenshots**
image

**Desktop (please complete the following information):**
 All browsers and operating systems but easiest to reproduce on Safari MacOS

**Smartphone (please complete the following information):**
All browsers and operating systems

**Additional context**
-

### LINKED ISSUE #312: [BUG] Recommendation sometimes not works on page refresh
**Describe the bug**
Recommendation text sometimes not appear after page refresh

**To Reproduce**
1. Go to https://cugetreg.com
2. Add courses if not added yet
3. Refresh page multiple times and observe recommendation text on each refresh

**Expected behavior**
Recommendation text should appear on each page refresh

**Screenshots**

**Desktop (please complete the following information):**
- Microsoft Edge
- Safari Mac (Easiest to reproduce)

**Smartphone (please complete the following information):**

**Additional context**
- Probably due to race condition between courses cart init and recommendation api call

### LINKED ISSUE #313: [BUG] Recommendation should only works when there are course on the cart
**Describe the bug**
Recommendation API was called when no courses in cart

**To Reproduce**
1. Go to https://cugetreg.com
2. Remove all courses from all semesters and studyProgram
3. Refresh page

**Expected behavior**
Recommendation should not appear at all

**Screenshots**

**Desktop (please complete the following information):**
- Microsoft edge but should happened in all device

**Smartphone (please complete the following information):**
 

**Additional context**

### PR DESCRIPTION
**Pull request by new5558**
## Why did you create this PR
- To fix the mentions bugs on recommendation

## What did you do
- Fix the following:
- Close #311 
- Close #312 
- Close #313

## Demo
[https://dev.cugetreg.com](https://dev.cugetreg.com) (COSINE variant)

## Checklist
- [X] Deploy a demo
- [ ] Check browsers compatibility
- [ ] Wrote coverage tests

### CODE REVIEW

File: src/modules/CourseSearch/components/RecommendationText/index.tsx [hunk 1]
```diff
@@ -99,10 +99,22 @@ const RecommendationText: React.FC<{ variant: string }> = observer((props: { var
...
+      studyProgram: item.studyProgram,
+      academicYear: item.academicYear,
+    },
+  }))
+
+  if (!selectedCourses || selectedCourses.length === 0) return null
+
+  if (recommendationVariant !== 'NONE')
+    return <RecommendationText variant={recommendationVariant} selectedCourses={selectedCourses} />
   else return <></>
```
Comment (saenyakorn) [inline src/modules/CourseSearch/components/RecommendationText/index.tsx:119]:
```suggestion
  if (recommendationVariant === 'NONE') return null
  
  return <RecommendationText variant={recommendationVariant} selectedCourses={selectedCourses} />
```

**Review by saenyakorn — APPROVED** [review]
LGTM

File: src/modules/CourseSearch/components/RecommendationText/index.tsx (hunk 1 above)
Comment (new5558) [reply src/modules/CourseSearch/components/RecommendationText/index.tsx]: Please check new one

**Comment by new5558** [comment]
FYI: Preview on dev branch is deploying

**Review by saengowp — APPROVED** [review]
Dev is down so I didn't check, It looks fine.

**Pull request merged**